TicketsDemo.Domain/DefaultImplementations/BookingPriceCalculationStrategy.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TicketsDemo.Domain.DTO;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Data.Repositories;
+using TicketsDemo.Domain.Interfaces;
+
+namespace TicketsDemo.Domain.DefaultImplementations
+{
+    public class BookingPriceCalculationStrategy : IPriceCalculationStrategy
+    {
+        private IRunRepository _runRepository;
+        private ITrainRepository _trainRepository;
+        private IRepresentativeRepository _representativeRepository;
+
+        public BookingPriceCalculationStrategy(
+            IRunRepository runRepository, 
```

Comment:
instead of pulling here IRunRepo abnd ITrainRepo and then duplicating logic from DefaultPriceCalculationStrategy - just pull here IPriceCalculationStrategy and use it's output